sklearn/utils/validation.py

```diff
@@ -1588,3 +1588,31 @@ def _check_fit_params(X, fit_params, indices=None):
             )
 
     return fit_params_validated
+
+
+def _get_feature_names(X):
+    """Get feature names from X.
+
+    Support for other array containers should place its implementation here.
+
+    Parameters
+    ----------
+    X : dataframe of shape (n_samples, n_features)
```

Comment:
```suggestion
    X : {dataframe-like} of shape (n_samples, n_features)
```
?

Reply:
I do not think there is a formal definition for "dataframe-like". The closest thing that exists is: https://github.com/data-apis/dataframe-api

Reply:
Then why do you have it in `_check_feature_names` in `base.py`? :D